src/rdiffbackup/locations/repository.py

```diff
@@ -114,18 +114,25 @@ def setup(self, src_dir=None, owners_map=None, action_name=None):
             else:
                 self._shadow = self.base_dir.conn._repo_shadow.RepoShadow
 
-            if not self.lock():
+            lock_result = self.lock()
+            if lock_result is False:
                 if self.force:
                     log.Log("Repository is locked by file {lf}, another "
                             "action is probably on-going. Enforcing anyway "
                             "at your own risk".format(lf=self.lockfile),
                             log.WARNING)
                 else:
                     log.Log("Repository is locked by file {lf}, another "
-                            "action is probably on-going. Either wait, remove "
-                            "the lock or use the --force option".format(
+                            "is probably on-going, or something went "
+                            "wrong. Either wait, remove the lock "
+                            "or use the --force option".format(
                                 lf=self.lockfile), log.ERROR)
                     return Globals.RET_CODE_ERR
+            elif lock_result is None:
+                log.Log("Repository couldn't be locked by file {lf}, probably "
```

Comment:
Wont' this message get printed on every run ? The lock file is deleted on success.

Otherwise I'm confused.

Reply:
No, unlock only removes the lock on the file, but doesn't remove the file https://github.com/rdiff-backup/rdiff-backup/blob/f8e4dd58af098a1812342e7b1df1deefdd939ea5/src/rdiffbackup/locations/_repo_shadow.py#L1154

Reply:
Thanks for the clarification.